src/Analyser/ExpressionContext.php

```diff
@@ -60,4 +61,14 @@ public function getInAssignRightSideNativeType(): ?Type
 		return $this->inAssignRightSideNativeType;
 	}
 
+	public function withOverriddenType(?Type $type): self
+	{
+		return new self($this->isDeep, $this->inAssignRightSideVariableName, $this->inAssignRightSideType, $this->inAssignRightSideNativeType, $type);
+	}
+
+	public function getOverriddenType(): ?Type
```

Comment:
Why this does have to be here actually at all? Why hasn't been the `passedToType` parameter kept? Why is this named "overriden type"? What is it overriding?